includes/class-wc-checkout.php

```diff
@@ -655,15 +655,15 @@ protected function validate_posted_data( &$data, &$errors ) {
 					$valid_states = WC()->countries->get_states( $country );
 
 					if ( ! empty( $valid_states ) && is_array( $valid_states ) && sizeof( $valid_states ) > 0 ) {
-						$valid_state_values = array_flip( array_map( 'wc_strtoupper', $valid_states ) );
+						$valid_state_values = array_map( 'wc_strtoupper', array_flip( array_map( 'wc_strtoupper', $valid_states ) ) );
 						$data[ $key ]       = wc_strtoupper( $data[ $key ] );
 
 						if ( isset( $valid_state_values[ $data[ $key ] ] ) ) {
 							// With this part we consider state value to be valid as well, convert it to the state key for the valid_states check below.
 							$data[ $key ] = $valid_state_values[ $data[ $key ] ];
 						}
 
-						if ( ! in_array( $data[ $key ], array_keys( $valid_states ) ) ) {
+						if ( ! in_array( $data[ $key ], array_values( $valid_state_values ) ) ) {
```

Comment:
`in_array()` already look for values by default, so no need of `array_values()`.